task/upload.go

```diff
@@ -480,19 +478,6 @@ func complementCatalystPipeline(tctx *TaskContext, assetSpec api.AssetSpec, call
 		}
 	}
 
-	if !FlagCatalystProbesFile {
```

Comment:
I'm almost sure there is one huge optimization we can do now: if catalyst copied the source file (`catalystCopiedSource`) and there is no IPFS storage (L481), we don't even need to download the source file anymore! Can probably do an early return around L447.

Haven't double checked alllll the logic that still remains here tho, so do make sure, but it seems overall possible 👀